app/src/main/java/com/flowfoundation/wallet/manager/backup/BackupCryptoProvider.kt

```
@@ -1,60 +1,223 @@
+                    val finalSignature = if (signature.size == 65) {
+                        logd("BackupCryptoProvider", "  Trimming recovery ID from 65-byte signature for domain signing")
+                        signature.copyOfRange(0, 64) // Remove the last byte (recovery ID)
+                    } else {
+                        logd("BackupCryptoProvider", "  Using signature as-is (${signature.size} bytes) for domain signing")
+                        signature
+                    }
+                    
+                    return finalSignature
+                } catch (e: Exception) {
+                    throw e
+                }
+            }
+            
+            override suspend fun signAsUser(bytes: ByteArray): ByteArray {
+                logd("BackupCryptoProvider", "KMM Signer.signAsUser() called")
+                return signWithDomain(bytes, DomainTag.User.bytes)
+            }
+            
+            override suspend fun signAsTransaction(bytes: ByteArray): ByteArray {
+                logd("BackupCryptoProvider", "KMM Signer.signAsTransaction() called")
+                return signWithDomain(bytes, DomainTag.Transaction.bytes)
+            }
+        }
     }
 
-    override fun getHashAlgorithm(): HashAlgorithm {
-        return HashAlgorithm.SHA2_256
+    /**
+     * Get the hashing algorithm used by this provider
+     * @return The configured hashing algorithm or a sensible default
+     */
+    override fun getHashAlgorithm(): HashingAlgorithm {
+        // Return the dynamically determined hashing algorithm if available
+        return hashingAlgorithm ?: run {
+            // Fallback to sensible defaults based on signing algorithm
+            when (signingAlgorithm) {
+                SigningAlgorithm.ECDSA_secp256k1 -> HashingAlgorithm.SHA2_256
+                SigningAlgorithm.ECDSA_P256 -> HashingAlgorithm.SHA3_256
```

The hash algorithm used for our multi-backup was previously SHA2-256. Please check if it can be synchronized with iOS.

---

@jaymengxy ok - should we always fallback to SHA2-256 if the hashing algorithm isn't found on the provider?

---

Another option would be to throw an error instead of falling back to a default algorithm

---

My question is why the SigningAlgorithm.ECDSA_P256 was changed to use the HashingAlgorithm.SHA3_256 instead of SHA2_256.

---

ah, I see what you mean. I think it is using SHA2_256, this is just a fallback in case the hashing algorithm isn't found on the provider. I can update it to SHA2_256 in the fallback.